packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx

```diff
@@ -0,0 +1,151 @@
+import { FilterableItem } from '@lightdash/common';
+import {
+    Highlight,
+    Loader,
+    MultiSelect,
+    MultiSelectProps,
+    Text,
+} from '@mantine/core';
+import { FC, ReactNode, useCallback, useMemo, useState } from 'react';
+
+import {
+    MAX_AUTOCOMPLETE_RESULTS,
+    useFieldValues,
+} from '../../../../../hooks/useFieldValues';
+import { useFiltersContext } from '../../FiltersProvider';
+import { mergeUniqueValues } from './autoCompleteUtils';
+
+type Props = Pick<MultiSelectProps, 'disabled' | 'placeholder'> & {
+    filterId: string;
+    field: FilterableItem;
+    values: string[];
+    suggestions: string[];
+    onChange: (values: string[]) => void;
+};
+
+const FilterMultiAutoComplete: FC<Props> = ({
+    filterId,
+    values,
+    field,
+    suggestions: initialSuggestionData,
+    disabled,
+    onChange,
+    placeholder,
+}) => {
+    const { projectUuid, getAutocompleteFilterGroup } = useFiltersContext();
+    if (!projectUuid) {
+        throw new Error('projectUuid is required in FiltersProvider');
+    }
+
+    const [search, setSearch] = useState('');
+
+    const autocompleteFilterGroup = useMemo(
+        () => getAutocompleteFilterGroup(filterId, field),
+        [field, filterId, getAutocompleteFilterGroup],
+    );
+
+    const { isLoading, results: resultsSet } = useFieldValues(
+        search,
+        initialSuggestionData,
+        projectUuid,
+        field,
+        autocompleteFilterGroup,
+        true,
+        { refetchOnMount: 'always' },
+    );
+
+    const results = useMemo(() => [...resultsSet], [resultsSet]);
+
+    const handleChange = useCallback(
+        (value: string[]) => {
+            onChange(value);
+        },
+        [onChange],
+    );
+
+    const handleAdd = useCallback(
+        (value: string) => {
+            onChange([...values, value]);
+            return value;
+        },
+        [onChange, values],
+    );
+
+    const handleOnClose = useCallback(() => {
+        setSearch('');
+    }, [setSearch]);
+
+    const data = useMemo(() => {
+        // Mantine does not show value tag if value is not found in data
+        // so we need to add it manually here
+        // also we are merging status indicator as a first item
```

Comment:
Isn't this what onCreate is for? This probably works too, but i think we usually add them to the data in onCreate. Which is maybe a little clearer and doesnt need an extra memo.

Reply:
so, `data` prop is basically `options` - what options show up in the dropdown. and `value` is for an array of selected options from data. if we have 

data of = [1,2,3]
and values of = [2,3,4]
then it only displays 2,3 as tags and omits 4

hope this explanation makes sense

Reply:
Ok, yea, I see. Thank you.